JavaScript/JavaScriptSDK/Sender.ts

```diff
@@ -200,14 +207,19 @@ module Microsoft.ApplicationInsights {
          * @param payload {string} - The data payload to be sent.
          * @param isAsync {boolean} - Indicates if the request should be sent asynchronously
          */
-        private _xhrSender(payload: string, isAsync: boolean, countOfItemsInPayload: number) {
+        private _xhrSender(payload: string[], isAsync: boolean, countOfItemsInPayload: number) {
+            // compose an array of payloads
+            var batch = this._buffer.batchPayloads();
+
             var xhr = new XMLHttpRequest();
             xhr[AjaxMonitor.DisabledPropertyName] = true;
             xhr.open("POST", this._config.endpointUrl(), isAsync);
             xhr.setRequestHeader("Content-type", "application/json");
-            xhr.onreadystatechange = () => Sender._xhrReadyStateChange(xhr, payload, countOfItemsInPayload);
-            xhr.onerror = (event: ErrorEvent) => Sender._onError(payload, xhr.responseText || xhr.response || "", event);
-            xhr.send(payload);
+            xhr.onreadystatechange = () => this._xhrReadyStateChange(xhr, payload, countOfItemsInPayload);
+            xhr.onerror = (event: ErrorEvent) => this._onError(payload, xhr.responseText || xhr.response || "", event);
```

Comment:
I'll move the `var batch = this._buffer.batchPayloads();` right before the send method, to make it less confusing.